index.js

```diff
@@ -765,15 +766,37 @@ _isntIgnored(path, stat) {
  * Provides a set of common helpers and properties relating to symlink and glob handling.
  * @param {Path} path file, directory, or glob pattern being watched
  * @param {Number=} depth at any depth > 0, this isn't a glob
- * @returns {WatchHelper} object containing helpers for this path
+ * @returns {Promise<WatchHelper>} object containing helpers for this path
  */
-_getWatchHelpers(path, depth) {
-  const watchPath = depth || this.options.disableGlobbing || !isGlob(path) ? path : globParent(path);
-  const follow = this.options.followSymlinks;
+async _getWatchHelpers(path, depth)
+{
+  let watchPath = depth || this.options.disableGlobbing || !isGlob(path) ? path : globParent(path);
 
+  const follow = this.options.followSymlinks;
+  watchPath=await this._mostSpecifiedExistingAncestor(watchPath);
   return new WatchHelper(path, watchPath, follow, this);
 }
 
+  /**
+   * Find the most recent (specified ) ancestor path that exists.
+   * @param {Path} path file or directory
+   * @returns {Promise<String>} the most recent ancestor path that exists
+   * @private
+   */
+  async _mostSpecifiedExistingAncestor(path)
+{
+  let father=globParent(path);
```

Comment:
You need to conform to existing code style. That means using constants, proper indentation etc.